.gitignore

```diff
@@ -3,3 +3,6 @@
 .packages
 .pub/
 pubspec.lock
+test/license_test_assets/license_scenarios
+.vscode
+third_party/spdx/licenses
```

Comment:
`third_party/spdx/licenses` and scenarios are excluded because they contain around 500 text files which will flood the whole PR and make it difficult to review I'll add them in a different PR.